src/js/utils/date-utils.js

```diff
@@ -0,0 +1,283 @@
+         *
+         * The input datetimeString can be just a date, a date and time,
+         * or a complex date/time string with TZ offset. This is fairly wide-open, as
+         * many strings will satisfy this requirement.
+         * (e.g. 01-01-16 and Jan 1, 2016, and 2016-01-01 01:00:00+00:00 will work)
+         *
+         * The recommendation is to use a fully informed string, e.g. '2016-01-01 01:00:00+00:00',
+         * to limit possible date, time, and timezone ambiguities.
+         *
+         * @param {string} datetimeString. A string representation of a datetime.
+         * @returns {object} A moment.js UTC datetime object.
+         *
+        */
+        dateObject = function(datetimeString) {
+            var utcDateObject;
+            if (isValid(datetimeString)) {
+                utcDateObject = moment(datetimeString).utc();
+            } else {
+                utcDateObject = '';
+            }
+            return utcDateObject;
+        };
+
+        /**
+         * Attempt to determine user's timezone from preferences, then browser settings.
+         * Will default to UTC.
+         *
+         * @param {string} timezone 'Region/Population_Center' e.g. 'America/New_York'.
+         * If this is None or undefined, we attempt to determine the TZ from system/browser settings.
+         * @return {string} Will attempt to return, in this order, the entered, determined, and default (UTC)
+         * timezone.
+         *
+         */
+        userTimezone = function(timezone) {
+            var determinedZone;
+            if (isValid(timezone)) {
+                determinedZone = timezone;
+            } else {
+                determinedZone = momentTZ.tz.guess();
+            }
+
+            if (isValid(determinedZone) === false) {
+                determinedZone = 'UTC';
+            }
+            return determinedZone;
+        };
+
+        /**
+         *
+         * Attempt to offset original time based on timezone.
+         *
+         * @param {Object} dateTimeObject A moment.js datetime object.
+         * @param {string} userTZ A timezone representation, e.g. 'America/New_York'.
+         * @return {object} A moment.js datetime object in a determined timezone.
+         */
+        localizeTime = function(dateTimeObject, userTZ) {
+            var localTime;
+            if (isValid(dateTimeObject) && isValid(userTZ)) {
+                localTime = dateTimeObject.tz(userTZ);
+            } else if (isValid(dateTimeObject)) {
+                localTime = dateTimeObject;
+            } else {
+                localTime = '';
+            }
+            return localTime;
+        };
+
+        /**
+         *
+         * Generate a timezone-aware and language-aware string.
+         *
+         * This includes a hook set of built-ins for standard formatting.
+         *
+         * Language is currently overridden by platform-set language preference. If no preference,
```

Comment:
Not sure why this says the language is "currently overridden"?  This makes me think that the requested language is never used and instead overridden by the one set in the browser.  However, that's not what the code seems to do.